File: Server/src/main/java/org/example/dto/product/ProductUpdateDTO.java
package org.example.dto.product;

import jakarta.annotation.Nullable;
import lombok.Data;
import org.springframework.web.multipart.MultipartFile;

@Data
public class ProductUpdateDTO {
    private String name;
    private double price;
    private String description;
    @Nullable
    private Long[] removeProductImages;
    @Nullable
    private MultipartFile[] newProductImages;
    private int categoryId;
}

File: Server/src/main/java/org/example/dto/product/ProductUpdateDTO.kt
package org.example.dto.product

import org.example.entities.CategoryEntity
import org.springframework.web.multipart.MultipartFile

class ProductUpdateDTO {
    lateinit var name: String
    var price = 0.0
    lateinit var description: String
    lateinit var oldProductImages: IntArray
    lateinit var newProductImages: Array<MultipartFile>
    lateinit var category: CategoryEntity
}